source/Core/InputValidator.php

```diff
@@ -174,37 +176,47 @@ public function checkEmail($user, $email)
      * Checking if user password is fine. In case of error
      * exception is thrown
      *
-     * @param User   $user                      Active user.
+     * @param string $email                     Email.
      * @param string $newPassword               New user password.
      * @param string $confirmationPassword      Retyped user password.
      * @param bool   $shouldCheckPasswordLength Option to check password length.
      *
      * @return Exception\StandardException|null
      */
-    public function checkPassword($user, $newPassword, $confirmationPassword, $shouldCheckPasswordLength = false)
+    public function checkPassword($email, string $newPassword, string $confirmationPassword, bool $shouldCheckPasswordLength = false)
```

Comment:
We cannot just simply change the parameter ;)

Reply:
Why not? `$user` doesn't seem to be used in the original method.  
<img width="989" alt="Bildschirmfoto 2020-08-24 um 16 50 01" src="https://user-images.githubusercontent.com/57272786/91059379-ec41bc00-e629-11ea-8632-5e6cf8820d4a.png">. 
https://github.com/OXID-eSales/oxideshop_ce/blob/master/source/Core/InputValidator.php#L184